lib/manifest-parser.js

```diff
@@ -2,11 +2,21 @@ var yaml = require('js-yaml')
 var yarn = require('@yarnpkg/lockfile')
 
 function splitNameAndVersion(nameAndVersion) {
-  var nameAndVersionWithoutLeadingAt = nameAndVersion.replace(/^@/, '')
+  // e.g. 'tweetnacl@^0.14.3, tweetnacl@~0.14.0' or '"tweetnacl@npm:^0.14.3, tweetnacl@npm:~0.14.0":'
+  var possibleMultipleRequirements = nameAndVersion.split(', ');
+  // only use first requirement's name, assuming that deps will always resolve from deps of the same name.
+  nameAndVersion = possibleMultipleRequirements[0];
 
-  var sep = nameAndVersionWithoutLeadingAt.split('@');
-  var name = sep[0];
-  var version = sep.slice(1, sep.length).join('@');
+  var nameAndVersionWithoutLeadingAt = nameAndVersion.replace(/^@/, '');
+
+  var name = nameAndVersionWithoutLeadingAt.split('@')[0];
+  var versions = [];
+  possibleMultipleRequirements.forEach((requirement) => {
+    var sep = requirement.split('@');
+    var version = sep.slice(1, sep.length).join('@').replace(/npm:/, ''); // e.g. "npm:^0.14.3"
+    versions.push(version);
+  })
+  var version = versions.join(' '); // comaparator intersection (https://classic.yarnpkg.com/lang/en/docs/dependency-versions/#toc-intersections)
```

Comment:
NB I'm assuming that `a, b:` means an "this satisfies both a and b", so I'm joining them with the intersection operator, i.e. " ".